v2/tools/generator/internal/astmodel/type_visitor_builder.go

```diff
@@ -10,11 +10,18 @@ import "fmt"
 // TypeVisitorBuilder provides a flexible way to create a TypeVisitor. Fields should be initialized
 // with funcs matching one of the following forms:
 //
-// func(this *TypeVisitor, it TypeName, ctx interface{}) (Type, error)
-// func(it TypeName) (Type, error)
-// func(it TypeName) Type
+// func(this *TypeVisitor, it <sometype>, ctx interface{}) (Type, error)
+// func(it <sometype>) (Type, error)
+// func(it <sometype>) Type
 //
-// These examples assume TypeName but can be generalized across any of the supported types.
+// o  Must always return Type, and optionally an error
+// o  <sometype> must match the type for the field being initialized
+//
+// Some examples:
+//
+// VisitTypeName = func(it TypeName) Type      // Works
```

Comment:
minor: Maybe include 1 more `// Works` example too at the bottom? Possibly that's one of the larger format ones like 
`func(it *ObjectType) (Type, error)` or `func(this *TypeVisitor, it *ObjectType, ctx interface{}) (Type, error)`. I think that helps accomplished what you were going for before of showing examples for `TypeName` _and_ other fields. One that is a ptr (since `TypeName` isn't) probably?

Reply:
Added.